pkgs/development/octave-modules/zeromq/default.nix

```diff
@@ -0,0 +1,28 @@
+{ buildOctavePackage
+, stdenv
+, fetchurl
+, zeromq
+}:
+
+buildOctavePackage rec {
+  pname = "zeromq";
+  version = "1.5.2";
+
+  src = fetchurl {
+    url = "https://octave.sourceforge.io/download.php?package=${pname}-${version}.tar.gz";
```

Comment:
Did you try using mirror://sourceforge instead?

Reply:
I have not. Is that a preferred way of downloading these tarballs? Or is that because this link actually redirects the `curl` command to `mirror://sourceforge`?

Reply:
> I have not. Is that a preferred way of downloading these tarballs?

For anything from sourceforge it is. It redirect the downloads to known good mirrors instead of choosing a random one. You can find the list in here https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchurl/mirrors.nix#L19.

> Or is that because this link actually redirects the curl command to mirror://sourceforge?

curl does not know anything from that. Nix builds the url together with a link from the mirrors.